From dd6954eea98810e480caad554c637925e92c971f Mon Sep 17 00:00:00 2001
From: Rajendra Nayak <rnayak@ti.com>
Date: Wed, 30 Jul 2014 22:41:42 -0500
Subject: [PATCH 0899/1587] ARM: AM43x: Add suspend/resume callbacks for
 clockevent

With gptimers being used as both the clockevent and clocksource
on AM43x, we need the suspend/resume callbacks populated to idle
the non wakeup domain gptimer in order to achieve system low
power state in suspend.

Signed-off-by: Rajendra Nayak <rnayak@ti.com>
Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 arch/arm/mach-omap2/timer.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/timer.c b/arch/arm/mach-omap2/timer.c
index 99448a4..64b16e5 100644
--- a/arch/arm/mach-omap2/timer.c
+++ b/arch/arm/mach-omap2/timer.c
@@ -401,7 +401,7 @@ static void __init omap2_gp_clockevent_init(int gptimer_id,
 					3, /* Timer internal resynch latency */
 					0xffffffff);
 
-	if (soc_is_am33xx()) {
+	if (soc_is_am33xx() || soc_is_am43xx()) {
 		clockevent_gpt.suspend = omap_clkevt_suspend;
 		clockevent_gpt.resume = omap_clkevt_resume;
 
-- 
1.7.5.4

